pootle/apps/pootle_store/updater.py

```diff
@@ -137,6 +137,30 @@ def should_update_target(self):
                 self.conflict_found
                 and self.update.resolve_conflict == POOTLE_WINS))
 
+    @property
+    def should_update_source(self):
+        return (
+            self.newunit
+            and self.source_updated)
+
+    @property
+    def should_update_state(self):
+        return (
+            self.newunit
+            and self.state_updated)
+
+    @cached_property
+    def source_updated(self):
+        if not self.newunit:
+            return False
+        return self.unit.source == self.newunit.source
+
+    @cached_property
+    def state_updated(self):
+        if not self.newunit:
+            return False
+        return self.unit.isfuzzy() != self.newunit.isfuzzy()
```

Comment:
`isobsolete` is covered by `should_unobsolete`, `istransalted` – by `target_changed`.
ideally it replaces this method https://github.com/translate/pootle/blob/c8b02dd1fd52829a5408eb2f69f777c8f8f2e1ae/pootle/apps/pootle_store/models.py#L412-L497.
though probs all those wrappers like `should_xxx` were added for optimizations...